p2p/p2p.md

```diff
@@ -0,0 +1,220 @@
+# P2P Design
+
+`p2p` package should expose minimal functions as a service to other components in Spacemesh.
+- `SendMessage(MSG, nodeID)`
+- `Broadcast(MSG)`
+- `OnRecv(msgtype, optional:peer, callback(func or chan))`  / `RegisterProtocol(Protocol, handler)`
+
```

Comment:
Consider adding a comment saying that we only support one "session" per protocol